src/hooks/useMultiSelect.ts

```diff
@@ -4,68 +4,54 @@ import {
   SelectedItems,
 } from "../components/MultiSelect/MultiSelect";
 
-interface useMultiSelectProps {
-  multiSelectId: string;
-  items: MultiSelectItem[];
-}
-
 /**
- * The useMultiSelect The hook returns an object containing all the props and state needed to handle the selectedItems.
+ * The useMultiSelect The hook returns an object containing all the props and state needed to handle the selectedItems in conjunction with one single Multiselect component as well as a Group of MultiSelects.
  * That includes the functions onChange, onClear, onMixedStateChange for handling any changes to the selection of items and the current state of the selection: selectedItems.
  */
-export default function useMultiSelect({
-  multiSelectId,
-  items,
-}: useMultiSelectProps) {
+
+export default function useMultiSelect() {
   const [selectedItems, setSelectedItems] = useState<SelectedItems | {}>({});
 
-  function handleChange(itemId: string) {
+  const handleChange = (itemId: string, multiSelectId: string) => {
     let itemIds;
-    // Check if the id already exists in the state.
-    if (selectedItems[multiSelectId] !== undefined) {
-      let itemIdExists =
-        selectedItems[multiSelectId].items.indexOf(itemId) > -1;
-      // Make a copy of the existing items array.
+    // Check if the multiSelecy already exists in the state.
+    if (selectedItems.hasOwnProperty(multiSelectId)) {
+      // Make a copy of the existing selected items.
       itemIds = selectedItems[multiSelectId].items.slice();
-      // If id exists, remove it from the array.
-      if (itemIdExists) {
+      // If itemId is in the selectedItems, remove it from the array.
+      if (selectedItems[multiSelectId].items.indexOf(itemId) > -1) {
         itemIds = itemIds.filter((id) => id !== itemId);
       } else {
         // Add it to the array, but modify the copy, not the original.
         itemIds.push(itemId);
       }
+      // If there were no items from the multiSelect in the selectedItems before
     } else {
+      // Add the itemId to the itemsIds
       itemIds = [];
       itemIds.push(itemId);
     }
-
+    // Update selectedItems on state to reflect the new selection
     setSelectedItems({
       ...selectedItems,
       [multiSelectId]: {
         items: itemIds,
       },
     });
-  }
+  };
 
-  /**
-   * handleMixedStateChange is only used for the "dialog" variant. It handles the state for checkbox options with child options.
-   * It accepts one argument - the id of the parent checkbox option.
-   */
```

Comment:
It would be nice to have comments for each function to explain what they do. This comment was fine before, why was it removed and not updated?